airbyte-integrations/connectors/destination-snowflake/src/main/kotlin/io/airbyte/integrations/destination/snowflake/operation/SnowflakeStagingClient.kt

```diff
@@ -144,14 +173,66 @@ class SnowflakeStagingClient(private val database: JdbcDatabase) {
         streamId: StreamId
     ) {
         try {
+            val queryId = UUID.randomUUID()
             val query = getCopyQuery(stageName, stagingPath, stagedFiles, streamId)
-            log.info { "Executing query: $query" }
-            database.execute(query)
+            log.info { "query $queryId, $query" }
+            // queryJsons is intentionally used here to get the error message in case of failure
+            // instead of execute
+            val results = database.queryJsons(query)
+            if (results.isNotEmpty()) {
+                // There will be only one row returned as the result of COPY INTO query
+                val copyResult = getCopyResult(results.first())
+                when (copyResult.copyStatus) {
+                    CopyStatus.LOADED ->
+                        log.info {
+                            "query $queryId, successfully loaded ${copyResult.rowsLoaded} rows of data into table"
+                        }
+                    CopyStatus.LOAD_FAILED -> {
+                        log.error {
+                            "query $queryId, failed to load data into table, " +
+                                "rows_parsed: ${copyResult.rowsParsed}, " +
+                                "rows_loaded: ${copyResult.rowsLoaded} " +
+                                "errors: ${copyResult.errorsSeen}, " +
+                                "firstError: ${copyResult.firstError}"
+                        }
+                        throw Exception(
+                            "COPY into table failed with ${copyResult.errorsSeen} errors, check logs"
+                        )
+                    }
+                    CopyStatus.UNKNOWN ->
+                        log.warn { "query $queryId, unrecognized result format, $results" }
```

Comment:
do we want to actually throw here? "we don't know if we successfully loaded the data" doesn't sound successful to me (so IMO we shouldn't e.g. ack state messages)